### PR TITLE
Use go-kit levels and its level filter

### PR DESCRIPTION
**Pull request by metalmatze**

### CODE REVIEW

**Comment by dominikschulz** [comment]
Well done, thanks so much!
Will test it tomorrow and if it works as expected I'll merge it.

File: main.go [hunk 1]
```diff
@@ -35,18 +35,27 @@ func main() {
...
+	logger = log.With(logger,
 		"ts", log.DefaultTimestampUTC,
+		"caller", log.DefaultCaller,
 	)
-	logger = leveled.NewFromEnv(logger)
-	logger = log.With(logger, "caller", log.DefaultCaller)
+	// set the allowed log level filter
+	switch strings.ToLower(os.Getenv("LOGLEVEL")) {
+	case "debug":
+		level.NewFilter(logger, level.AllowDebug())
```
Comment (dominikschulz) [inline main.go]: Small mistake: You need to assing the return value of level.NewFilter to the `logger` variable or else it has no effect.

File: main.go [hunk 2]
```diff
@@ -35,18 +35,27 @@ func main() {
...
+	case "warn":
+		level.NewFilter(logger, level.AllowWarn())
```
Comment (dominikschulz) [inline main.go]: Same here

File: main.go [hunk 3]
```diff
@@ -35,18 +35,27 @@ func main() {
...
+	case "error":
+		level.NewFilter(logger, level.AllowError())
```
Comment (dominikschulz) [inline main.go]: and here

File: main.go [hunk 4]
```diff
@@ -35,18 +35,27 @@ func main() {
...
+	default:
+		level.NewFilter(logger, level.AllowInfo())
```
Comment (dominikschulz) [inline main.go]: and here.

Also I'd like to change the default level to `level.AllowAll()` and introduce another case for `info`.

**Comment by metalmatze** [comment]
Whoops. 🤣 I will improve it tomorrow.

**Pull request merged**